catalyst/metrics/_segmentation.py

```diff
@@ -151,11 +152,30 @@ def compute_key_value(self) -> Dict[str, torch.Tensor]:
         Returns:
              dict of metrics, including micro, macro and weighted (if weights were given) metrics
         """
+        if self.weights is not None:
+            per_class, micro_metric, macro_metric, weighted_metric = self.compute()
+        else:
+            per_class, micro_metric, macro_metric = self.compute()
```

Comment:
let's always return `per_class, micro_metric, macro_metric, weighted_metric` and make `weighted_metric` if `self.weights is None`